### ISSUE TITLE
Post Comments Block: Implementation Notes

### ISSUE THREAD

**Issue by ockham** [WordPress/gutenberg#24101]
The Post Comments Block is currently implemented to be [rendered](https://github.com/WordPress/gutenberg/blob/d86cd5f0b36b3de3034096c09e0aaf7509367c4c/packages/block-library/src/post-comments/index.php#L31) by [`comments_template()`](https://developer.wordpress.org/reference/functions/comments_template/) on the frontend (which can be fairly complex, containing e.g. the comment author name and gravatar, date, and content).

In the editor, it is a [loop over paragraphs containing individual comments](https://github.com/WordPress/gutenberg/blob/d86cd5f0b36b3de3034096c09e0aaf7509367c4c/packages/block-library/src/post-comments/edit.js#L20-L22).

We need match what is editable in the editor with the output produced by the frontend. Furthermore, before making any edits, both needs to be determined by the theme. To that end, we need to apply a similar paradigm to Post Comments that we're already applying to the rest of a theme in FSE. Specifically, we need to:

- Use the [`comments_template` hook](https://developer.wordpress.org/reference/hooks/comments_template/) to use a block-based system rather than a PHP one.
- (To be continued)

**Comment by karmatosed** [comment]
Unless I am mistaken this doesn't need design, so for now going to remove the label. It can always be added back in though.

**Comment by jameskoster** [comment]
If we intend to mimic the behaviour of the Query block (I agree we should) then I am wondering what purpose the "Post Comment" block serves in this context? There isn't an equivalent block for queries/posts.

---

>  Comment Author
>  Comment Date
>  Comment Content

In addition to these, we may also need:

* Comment Author Gravatar
* Comment Reply Link
* Edit Comment link 

---

These new blocks will need design for their placeholders/previews, descriptions, and icons. I created a Figma file [here](https://www.figma.com/file/AagoBClTm0AgpfYCSKQl0m/Comment-blocks?node-id=0%3A1) for us to explore this.

**Comment by jameskoster** [comment]
I shared designs for the new blocks we need in separate issues. Links in the OP :)

**Comment by carolinan** [comment]
> If we intend to mimic the behaviour of the Query block (I agree we should) then I am wondering what purpose the "Post Comment" block serves in this context? There isn't an equivalent block for queries/posts.
> 
> > Comment Author
> > Comment Date
> > Comment Content
> 
> In addition to these, we may also need:
> 
> * Comment Author Gravatar
> * Comment Reply Link
> * Edit Comment link
> 
> These new blocks will need design for their placeholders/previews, descriptions, and icons. I created a Figma file [here](https://www.figma.com/file/AagoBClTm0AgpfYCSKQl0m/Comment-blocks?node-id=0%3A1) for us to explore this.

I think the post comment block has a different purpose because it only shows one specific comment and this comment can be unrelated to the current post/page/view.

**Comment by jameskoster** [comment]
Yes, I think that block has more utility in the post editor. But even there it feels kind of niche.

**Comment by carolinan** [comment]
We also have to consider the different settings that are available in the admin area under Settings > Discussion such as the nested comments and the comment pagination.

**Comment by jameskoster** [comment]
That is true, we may also need new blocks for the pagination. Hopefully we can lean on the query pagination blocks there, seeing as they serve (roughly) the same purpose, and will have crossed a similar hurdle when accounting for the posts_per_page setting.

**Comment by SantosGuillamot** [comment]
Hi there! Really nice work done here 👏  

I'd like to retake it and create a Post Comments Loop block, similar to the current [Query Loop block](https://wordpress.org/support/article/query-loop-block/), to display all the comments of a specific posts. I've created [this tracking issue](https://github.com/WordPress/gutenberg/issues/34994) with a detailed explanation and all the steps needed for anyone interested. I've also created [this project](https://github.com/WordPress/gutenberg/projects/61) to keep track of all the relevant issues.

**Comment by mtias** [comment]
Great, let's consolidate in #34994!
